src/include/miopen/fusion.hpp

```diff
@@ -52,13 +58,15 @@ enum FusionKernelSourceType
 
 struct OperatorArgs : miopenOperatorArgs
 {
-    OperatorArgs();
-    void ins_arg(std::string name, OpKernelArg v);
+    std::vector<std::shared_ptr<fusion::FusionOpInvokeParamBase>> params;
     friend std::ostream& operator<<(std::ostream& stream, const OperatorArgs& x);
-    std::vector<OpKernelArg> args_vec;
-    std::unordered_map<std::string, OpKernelArg> args_map;
+    void SetArg(size_t idx, const std::shared_ptr<fusion::FusionOpInvokeParamBase>& arg)
+    {
+        if(params.size() < (idx + 1))
+            params.resize(idx + 1);
+        params[idx] = arg;
+    }
```

Comment:
```suggestion
    void SetArg(size_t idx, std::shared_ptr<fusion::FusionOpInvokeParamBase> arg)
    {
        if(params.size() < (idx + 1))
            params.resize(idx + 1);
        params[idx] = std::move(arg);
    }
```
This allows both move and copy. Also one may want to enforce move here with &&. And I'd rather use unique_ptr here anyway.

Reply:
Changed to unique_ptr